src/Analyser/NodeScopeResolver.php

```diff
@@ -3863,7 +3863,12 @@ private function processStmtVarAnnotation(MutatingScope $scope, Node\Stmt $stmt,
 					$certainty = TrinaryLogic::createYes();
 				}
 
-				$scope = $scope->assignVariable($name, $varTag->getType(), new MixedType(), $certainty);
+				$scope = $scope->assignVariable(
+					$name,
+					$varTag->getType(),
+					$scope->getNativeType(new Variable($name)),
```

Comment:
since this internally falls back to `getType()` it could be potentially wrong, right? is there any better way to get the native (variable/expression) type then?

Reply:
I think there are no way now.
We have to finish https://github.com/phpstan/phpstan-src/pull/1943 to make it correct.